src/app/grid/grid-groupby-custom-sample/grid-groupby-custom-sample.component.ts

```diff
@@ -0,0 +1,137 @@
+import { Component, ViewChild, ViewEncapsulation } from "@angular/core";
+import {
+    DefaultSortingStrategy,
+    IgxGridComponent,
+    SortingDirection
+} from "igniteui-angular";
+import { INVOICE_DATA } from "../../data/invoiceData";
+
+@Component({
+    encapsulation: ViewEncapsulation.None,
+    selector: "app-grid-groupby-custom-sample",
+    styleUrls: ["./grid-groupby-custom-sample.component.scss"],
+    templateUrl: "./grid-groupby-custom-sample.component.html"
+})
+export class GridGroupByCustomSampleComponent {
+    @ViewChild("grid1", { read: IgxGridComponent, static: true })
+    public grid1: IgxGridComponent;
+    public data;
+    public sortingStrategy;
+    public groupByOptions = [
+        { name: "Day", ref: DaySortingStrategy.instance() },
+        { name: "Week", ref: WeekSortingStrategy.instance() },
+        { name: "Month", ref: BaseSortingStrategy.instance() },
+        { name: "Year", ref: BaseSortingStrategy.instance() }
+    ];
+
+    public groupByMode = this.groupByOptions[0].name;
+    public initialExpr;
+
+    constructor() {
+        this.data = INVOICE_DATA;
+        this.sortingStrategy = this.groupByOptions[0].ref;
+
+        this.initialExpr = [
+            {
+                dir: SortingDirection.Asc,
+                fieldName: "OrderDate",
+                ignoreCase: true,
+                strategy: this.sortingStrategy,
+                groupingComparer: (a, b) => {
+                    const dateA = this.sortingStrategy.getParsedDate(a);
+                    const dateB = this.sortingStrategy.getParsedDate(b);
+                    if (this.groupByMode === 'Month') {
+                        return dateA.month === dateB.month ? 0 : -1;
```

Comment:
Wouldn't this group values with the same month that are in different years?

Reply:
No, because firstly the sorting strategy take care of sorting the records by year and month. The comparer is called over the sorted data.